In [1]:
%load_ext sql

<h2>1. Analyzing The World Bank's International Debt Data</h2>

<p>Countries may take on debt to manage their economy and provide infrastructure spending for their citizens. <a href="https://www.worldbank.org">The World Bank</a> provides debt to many developing countries, and in this notebook, we'll be analyzing international debt data collected by The World Bank.</p>

<p>The dataset includes information on the amount of debt (in USD) owed by developing countries across several categories. We'll answer questions like:</p>
<ul>
  <li>What is the total amount of debt owed by the countries listed in the dataset?</li>
  <li>Which country owes the most debt, and how much do they owe?</li>
  <li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>

<p


In [8]:
%%sql
SELECT *
FROM international_debt
LIMIT 5;

 * postgresql://prosperadudata:***@localhost/sql_practice
5 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001'))]

<h2>2. Finding the number of distinct countries</h2>

<p>Although the first five rows of our dataset provide information about the amount of debt owed by Afghanistan across various debt indicators, we don't yet know the number of distinct countries in the table. Since countries can have debt in multiple categories, there may be repeated entries for each country. Therefore, before performing our statistical analysis, we need to determine the number of unique countries in the dataset.</p>

<p>We'll accomplish this by running a SQL query that selects the distinct country names from the <code>international_debt</code> table and counts the number of unique entries. This will give us the total number of distinct countries in our dataset.</p>


In [11]:
%%sql
SELECT count(DISTINCT country_name) AS number_of_indebted_countries
FROM international_debt;

 * postgresql://prosperadudata:***@localhost/sql_practice
1 rows affected.


[(124,)]

<h2>3. Finding the distinct debt indicators</h2>

<p>With a count of 124 countries in our dataset, we now turn our attention to the different debt indicators. The <code>indicator_name</code> column briefly describes the purpose of each loan, while the <code>indicator_code</code> column represents the category of the debt.</p>

<p>Understanding these debt categories is important as it can provide insight into the specific areas where a country may be indebted. We'll extract the unique debt indicators present in our dataset by running a SQL query that selects the distinct <code>indicator_code</code> values from the <code>international_debt</code> table. This will allow us to analyze the data in a more meaningful way.</p>


In [22]:
%%sql
SELECT indicator_code, indicator_name
FROM international_debt
GROUP BY indicator_code, indicator_name
ORDER BY indicator_code

 * postgresql://prosperadudata:***@localhost/sql_practice
25 rows affected.


[('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)'),
 ('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)'),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)'),
 ('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)'),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)'),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)'),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)'),
 ('DT.AMT.PROP.CD', 'PPG, other private creditors (AMT, current US$)'),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)'),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)'),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)'),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)'),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)'),
 ('DT.DIS.PCBK.CD', 'PPG, commercial banks (DIS, current US$)'),
 ('DT.DIS.PROP.CD', 'PPG, other private creditors (DIS, current US$)'),
 ('DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)'),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)'),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)'),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)'),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)'),
 ('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)'),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)'),
 ('DT.INT.PCBK.CD', 'PPG, commercial banks (INT, current US$)'),
 ('DT.INT.PROP.CD', 'PPG, other private creditors (INT, current US$)'),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)')]

<h2>4. Totaling the amount of debt owed by the countries</h2>

<p>So far, we've analyzed the different debt indicators and the number of countries in our dataset. But what about the total amount of debt owed by all the countries in our dataset? This information can give us insight into the overall state of the global economy.</p>

<p>In this section, we'll focus on calculating the total amount of debt (in USD) owed by the different countries. We'll do this by running a SQL query that sums the debt values for each country </p>


In [24]:
%%sql
SELECT ROUND(sum(debt)) AS total_world_debt
FROm international_debt

 * postgresql://prosperadudata:***@localhost/sql_practice
1 rows affected.


[(Decimal('3079734487676'),)]

<h2>5. Country with the highest debt</h2>

<p>Now that we have the total amount of debt (in USD) owed by all the countries, let's find out which country owns the highest amount of debt and the amount owed. It's important to note that this debt is the sum of different debts owed by a country across several categories. Knowing the country with the highest debt can give us insight into its overall socio-economic situation.</p>

<p>To do this, we'll run a SQL query that selects the top 3 countries with the highest amount of debt.</p>


In [37]:
%%sql
SELECT country_name, ROUND(sum(debt)) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 3;


 * postgresql://prosperadudata:***@localhost/sql_practice
3 rows affected.


[('China', Decimal('285793494734')),
 ('Brazil', Decimal('280623966141')),
 ('South Asia', Decimal('247608723991'))]

## 6. Total amount of Debt <em>Nigeria<em> owes in this Dataset

<p>This query isn't related to this data story, but I'm Nigerian and just had to scratch my curiosity itch! 😂</p>


In [67]:
%%sql
SELECT country_name, ROUND(sum(debt)) AS total_debt
FROM international_debt
WHERE country_name = 'Nigeria'
GROUP BY country_name;


 * postgresql://prosperadudata:***@localhost/sql_practice
1 rows affected.


[('Nigeria', Decimal('14149905241'))]

## 7. Average amount of debt across indicators
<p>So, it was <em>China</em>, Followed by <em>Brazil</em> and <em>South Asia</em></p>
<p>We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of the different debt indicators in which the countries owe their debts. We can dig even further to find out on an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators.</p>

In [39]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    ROUND(avg(debt)) as average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt desc
LIMIT 10;

 * postgresql://prosperadudata:***@localhost/sql_practice
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194334')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041217')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983453')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694102')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024068')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623948'))]

## 8. The highest amount of principal repayments
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital.</p>
<p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>
<p>We can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [47]:
%%sql
SELECT country_name, indicator_code, indicator_name, ROUND(sum(debt)) AS average_debt
FROM international_debt
WHERE indicator_code = 'DT.AMT.DLXF.CD' 
GROUP BY country_name, indicator_code, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql://prosperadudata:***@localhost/sql_practice
10 rows affected.


[('China', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('96218620836')),
 ('Brazil', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('90041840304')),
 ('Russian Federation', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('66589761834')),
 ('Turkey', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('51555031006')),
 ('South Asia', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('48756295898')),
 ('India', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('31923507001')),
 ('Indonesia', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('30916112654')),
 ('Kazakhstan', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('27482093686')),
 ('Mexico', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('25218503927')),
 ('Least developed countries: UN classification', 'DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('25197029299'))]

## 9. The most common debt indicator
<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category</p>
<p>We saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out. </p>

In [52]:
%%sql
SELECT indicator_code, indicator_name, COUNT(indicator_code) as indicator_count
FROM international_debt
GROUP BY indicator_code, indicator_name
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;

 * postgresql://prosperadudata:***@localhost/sql_practice
20 rows affected.


[('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)', 124),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)', 124),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', 124),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', 124),
 ('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)', 124),
 ('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', 124),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', 123),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)', 122),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', 122),
 ('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)', 122),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)', 120),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', 113),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)', 98),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', 98),
 ('DT.INT.PCBK.CD', 'PPG, commercial banks (INT, current US$)', 84),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)', 84),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 79),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 79),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)', 69),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', 69)]

## 10. Other viable debt issues and conclusion
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue. But that is not the end of the story, a part of the story rather. </p>
<p>Let's change tracks from <code>debt_indicator</code>s now and focus on the amount of debt again. Let's find out the maximum amount of debt across the indicators along with the respective country names. With this, we will be in a position to identify the other plausible economic issues a country might be going through. By the end of this section, we will have found out the debt indicators in which a country owes its highest debt. </p>
<p>In this notebook, we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures. We also validated our findings to make sure the investigations are correct.</p>

In [62]:
%%sql
SELECT country_name, indicator_code, ROUND(max(debt)) AS maximum_debt
FROM international_debt
GROUP BY country_name, indicator_code
ORDER BY maximum_debt DESC
LIMIT 10;

 * postgresql://prosperadudata:***@localhost/sql_practice
10 rows affected.


[('China', 'DT.AMT.DLXF.CD', Decimal('96218620836')),
 ('Brazil', 'DT.AMT.DLXF.CD', Decimal('90041840304')),
 ('China', 'DT.AMT.DPNG.CD', Decimal('72392986214')),
 ('Russian Federation', 'DT.AMT.DLXF.CD', Decimal('66589761834')),
 ('Turkey', 'DT.AMT.DLXF.CD', Decimal('51555031006')),
 ('South Asia', 'DT.AMT.DLXF.CD', Decimal('48756295898')),
 ('Brazil', 'DT.AMT.PRVT.CD', Decimal('43598697499')),
 ('Russian Federation', 'DT.AMT.DPNG.CD', Decimal('42800154975')),
 ('Brazil', 'DT.AMT.DPNG.CD', Decimal('41831444053')),
 ('Least developed countries: UN classification', 'DT.DIS.DLXF.CD', Decimal('40160766262'))]